File: python/881. Boats to Save People.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        count, i, j = 0, 0, len(people)-1
        people.sort()
        while i <= j:
            if i != j and people[i]+people[j] <= limit:
                i += 1
                j -= 1
            else:
                j -= 1
            count += 1
        return count

File: go/881. Boats to Save People.go
func numRescueBoats(people []int, limit int) int {
	i, j, count := 0, len(people)-1, 0
	sort.Ints(people)
	for i <= j {
		if i != j && people[i]+people[j] <= limit {
			i += 1
		}
		j -= 1
		count += 1
	}
	return count
}